bin/add_transform_test.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+"""Automatically create transform tests input and output files given an input template.
+
+Usage:
+    add_transform_test.py --template-file=sam-template.yaml [--disable-api-configuration]
+    add_transform_test.py --template-file=sam-template.yaml
+
+Options:
+    --template-file=<i>             Location of SAM template to transform [default: template.yaml].
+    --disable-api-configuration     Disable adding REGIONAL configuration to AWS::ApiGateway::RestApi
+"""
+import json
+import subprocess
+import re
+import os
+import shutil
+import yaml
+import tempfile
+from docopt import docopt
+from pathlib import Path
+from typing import Any
+
+from samtranslator.yaml_helper import yaml_parse
+
+cwd = os.getcwd()
+CLI_OPTIONS = docopt(__doc__)
+
+
+def get_input_file_path() -> str:
+    input_file_option = CLI_OPTIONS.get("--template-file")
+    return os.path.join(cwd, input_file_option)
+
+
+def copy_input_file_to_transform_test_dir(input_file_path: str, transform_test_input_path: str) -> None:
+    shutil.copyfile(input_file_path, transform_test_input_path)
+
+    replace_arn_partitions(transform_test_input_path)
+    print(f"Transform Test input file generated {transform_test_input_path}")
+
+
+def add_regional_endpoint_configuration_if_needed(temp_output_file_path: str) -> None:
+    with open(temp_output_file_path, "r") as f:
+        sam_template = json.load(f)
+
+    for logical_id, resource in sam_template["Resources"].items():
+        if resource["Type"] == "AWS::ApiGateway::RestApi":
+            if "EndpointConfiguration" not in resource["Properties"]:
+                sam_template["Resources"][logical_id]["Properties"]["EndpointConfiguration"] = {"Types": ["REGIONAL"]}
+            if "Parameters" not in resource["Properties"]:
+                sam_template["Resources"][logical_id]["Properties"]["Parameters"] = {
+                    "endpointConfigurationTypes": "REGIONAL"
+                }
+
+    with open(temp_output_file_path, "w") as f:
+        json.dump(sam_template, f, indent=2)
+
+
+def generate_transform_test_output_files(input_file_path: str, file_basename: str) -> None:
+    output_file_option = file_basename + ".json"
+
+    # run sam-translate.py and get the temporary output file
+    with tempfile.NamedTemporaryFile() as temp_output_file:
+        subprocess.run(
+            [
+                "python",
+                cwd + "/bin/sam-translate.py",
+                "--template-file",
+                input_file_path,
+                "--output-template",
+                temp_output_file.name,
+            ]
+        )
```

Comment:
So it fails if command fails:

```suggestion
            ],
            check=True,
        )
```